src/main/kotlin/com/terraformation/backend/accelerator/api/ApplicationsController.kt

```diff
@@ -0,0 +1,185 @@
+  ): ListApplicationsResponsePayload {
+    TODO()
+  }
+
+  @Operation(
+      summary = "Restart a previously-submitted application",
+      description = "If the application has not been submitted yet, this is a no-op.")
+  @PostMapping("/{applicationId}/restart")
+  fun restartApplication(@PathVariable applicationId: ApplicationId): SimpleSuccessResponsePayload {
+    TODO()
+  }
+
+  @Operation(
+      summary = "Submit an application for review",
+      description = "If the application has already been submitted, this is a no-op.")
+  @PostMapping("/{applicationId}/submit")
+  fun submitApplication(@PathVariable applicationId: ApplicationId): SimpleSuccessResponsePayload {
+    TODO()
+  }
+
+  @InternalEndpoint
+  @Operation(
+      summary = "Update an application's metadata to reflect a review",
+      description = "This is an internal-user-only operation.")
+  @PostMapping("/{applicationId}/review")
+  @RequireGlobalRole([GlobalRole.AcceleratorAdmin, GlobalRole.SuperAdmin, GlobalRole.TFExpert])
+  fun reviewApplication(
+      @PathVariable applicationId: ApplicationId,
+      @RequestBody payload: ReviewApplicationRequestPayload
+  ): SimpleSuccessResponsePayload {
+    TODO()
+  }
+
+  @Operation(summary = "Update an application's boundary")
+  @PutMapping("/{applicationId}/boundary")
+  fun updateApplicationBoundary(
+      @PathVariable applicationId: ApplicationId,
+      @RequestBody payload: UpdateApplicationBoundaryRequestPayload
+  ): SimpleSuccessResponsePayload {
+    TODO()
+  }
+
+  @Operation(summary = "Update an application's boundary using an uploaded file")
+  @PostMapping("/{applicationId}/boundary", consumes = [MediaType.MULTIPART_FORM_DATA_VALUE])
+  fun uploadApplicationBoundary(
+      @PathVariable applicationId: ApplicationId,
+      @RequestPart file: MultipartFile
+  ): SimpleSuccessResponsePayload {
+    TODO()
+  }
+}
+
+data class ApplicationHistoryPayload(
+    val feedback: String?,
+    @Schema(
+        description =
+            "Internal-only comment, if any. Only set if the current user is an internal user.")
+    val internalComment: String?,
+    val modifiedTime: Instant,
+    val status: ApplicationStatus,
+)
+
+data class ApplicationPayload(
+    val boundary: Geometry?,
+    val createdTime: Instant,
+    val feedback: String?,
+    val id: ApplicationId,
+    @Schema(
+        description =
+            "Internal-only comment, if any. Only set if the current user is an internal user.")
+    val internalComment: String?,
+    @Schema(
+        description =
+            "Internal-only reference name of application. Only set if the current user is an " +
+                "internal user.")
+    val internalName: String?,
+    val organizationId: OrganizationId,
+    val projectId: ProjectId,
+    val status: ApplicationStatus,
+)
+
+data class CreateApplicationRequestPayload(
+    @Schema(oneOf = [MultiPolygon::class, Polygon::class]) //
+    val boundary: Geometry?,
```

Comment:
I know this is optional but I think it is very unlikely we use this param here. OK to leave it in, in case we do support this in the FE at some point.